text/0046-404-405-http-error-code.md

```diff
@@ -0,0 +1,121 @@
+|-----------------------------------------------------|----------------|-----------------------------------------|--------------------|
+| 404 HTTP response status code                       |       x        |                    x                    |         x          |
+| 405 HTTP response status code                       |                |                                         |                    |
+| Presence of a well formed response format           |                |                    x                    |                    |
+
+| 0.21 Behaviors                                      | Invalid url    | Valid url but object not found          | Method not allowed |
+|-----------------------------------------------------|----------------|-----------------------------------------|--------------------|
+| 404 HTTP response status code                       |       x        |                                         |                    |
+| 405 HTTP response status code                       |                |                                         |          x         |
+| 400 HTTP response status code                       |                |                    x                    |                    |
+| Presence of a well formed reponse format            |                |                                         |                    |
+
+The next parts will list the desired behaviors for the release of version 0.21.
+
+| 0.21 Behaviors proposal                             | Invalid url    | Valid url but object not found          | Method not allowed |
+|-----------------------------------------------------|----------------|-----------------------------------------|--------------------|
+| 404 HTTP response status code                       |       x        |                     x                   |                    |
+| 405 HTTP response status code                       |                |                                         |          x         |
+| Presence of a well formed reponse format            |       x        |                     x                   |          x         |
+
+#### 404 Code
+
+The API must return a `404` error code when the resource does not exist or when the url used does not allow to reach a resource.
+
+E.g. Asking for an inexistent ressource.
+
+**GET `/indexes/:index_uid/documents/2`**
+
+In that case, the response body contain a `document_not_found` errorCode.
+
+E.g.
+
+```
+{
+    "message": "Document with id 2 not found",
+    "errorCode": "document_not_found",
+    "errorType": "invalid_request_error",
+    "errorLink": "https://docs.meilisearch.com/errors#document_not_found"
+}
+```
+
+E.g. Using a wrong url.
+
+**GET `/indexess/documents/2`**
+
+In that case, the response body should contain a `unrecognized_request_url` errorCode.
+
+```
+{
+    "message": "Unrecognized request URL (GET: /indexess/documents/2). See https://docs.meilisearch.com/reference/api",
```

Comment:
Not sure the message needs to refer to another link than `errorLink` in the message. The docs link should be explicit and re-link to the right resources.

Reply:
On this specific error, I can't think of a specific link! I was inspired by the error messages of Stripe API for this type of case which offers quite clear messages.